TombEngine/Game/Lara/lara_objects.cpp

```diff
@@ -258,46 +258,161 @@ void lara_as_pulley(ItemInfo* item, CollisionInfo* coll)
-	const auto& barItem = g_Level.Items[lara->Context.InteractedItem];
+	auto& player = GetLaraInfo(*item);
+
+	if (IsHeld(In::Action))
+	{
+		if (IsHeld(In::Forward))
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_SWING_START;
+		}
+		else if (IsHeld(In::Roll) || IsHeld(In::Back))
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_IDLE_TURN_180;
+		}
+
+		player.Control.HandStatus = HandStatus::Busy;
+		return;
+	}
+
+	player.Control.HandStatus = HandStatus::Free;
+	item->Animation.TargetState = LS_JUMP_BACK;
+}
+
+// State:	  LS_HORIZONTAL_BAR_IDLE (191)
+// Collision: lara_default_col()
+void lara_as_horizontal_bar_idle(ItemInfo* item, CollisionInfo* coll)
+{
+	auto& player = GetLaraInfo(*item);
+
+	if (IsHeld(In::Action))
+	{
+		if (IsHeld(In::Forward))
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_SWING_START;
+		}
+		else if (IsHeld(In::Roll) || IsHeld(In::Back))
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_IDLE_TURN_180;
+		}
+
+		player.Control.HandStatus = HandStatus::Busy;
+		return;
+	}
+
+	player.Control.HandStatus = HandStatus::Free;
+	item->Animation.TargetState = LS_JUMP_BACK;
+}
+
+// State:	  LS_HORIZONTAL_BAR_SWING_START (192)
+// Collision: lara_default_col()
+void lara_as_horizontal_bar_swing_start(ItemInfo* item, CollisionInfo* coll)
+{
+	auto& player = GetLaraInfo(*item);
+
+	if (IsHeld(In::Action))
+	{
+		if (IsHeld(In::Forward))
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_SWING;
+		}
+		else
+		{
+			item->Animation.TargetState = LS_HORIZONTAL_BAR_SWING_STOP;
+		}
+
+		player.Control.HandStatus = HandStatus::Busy;
+		return;
+	}
+
+	player.Control.HandStatus = HandStatus::Free;
+	item->Animation.TargetState = LS_JUMP_BACK;
+}
+
+// State:	  LS_HORIZONTAL_BAR_IDLE_TURN_180 (193)
+// Collision: lara_default_col()
+void lara_as_horizontal_bar_idle_turn_180(ItemInfo* item, CollisionInfo* coll)
+{
+	auto& player = GetLaraInfo(*item);
+
+	if (IsHeld(In::Action))
+	{
+		player.Control.HandStatus = HandStatus::Busy;
+		item->Animation.TargetState = LS_HORIZONTAL_BAR_IDLE;
+		return;
+	}
+
+	player.Control.HandStatus = HandStatus::Free;
+	item->Animation.TargetState = LS_JUMP_BACK;
+}
+
+// State:	  LS_HORIZONTAL_BAR_SWING_TURN_180 (194)
+// Collision: lara_default_col()
+void lara_as_horizontal_bar_swing_turn_180(ItemInfo* item, CollisionInfo* coll)
+{
+	auto& player = GetLaraInfo(*item);
+
+	player.Control.HandStatus = HandStatus::Busy;
+}
+
+// State:	  LS_HORIZONTAL_BAR_JUMP (129)
+// Collision: lara_default_col()
+void lara_as_horizontal_bar_jump(ItemInfo* item, CollisionInfo* coll)
+{
+	auto& player = GetLaraInfo(*item);
+	const auto& barItem = g_Level.Items[player.Context.InteractedItem];
 
 	item->Animation.IsAirborne = true;
 
 	if (item->Animation.FrameNumber == GetAnimData(*item).frameBase)
 	{
-		int distance = 0;
+		int dist = 0;
 		if (item->Pose.Orientation.y == barItem.Pose.Orientation.y)
-			distance = (barItem.TriggerFlags / 100) - 2;
```

Comment:
I didn't reach to understand the maths for the distance.
What value shall I put in the OCB to make Lara jump 1.5 sector, or 4 sectors?
and
Why does the Lara orientation matters for the else case? Does it really reaches to happens?

I know you didn't touch that but, could this be a chance to consider a new method?